### PR TITLE
Fix for Remote Rendering build failure

### PR DESCRIPTION
**Pull request by ChristopherManthei**
* This fixes an issue with Netty version resolution.

# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/remoterendering/azure-mixedreality-remoterendering/pom.xml [hunk 1]
```diff
@@ -94,6 +94,12 @@
       <version>1.12.0</version> <!-- {x-version-update;com.azure:azure-identity;dependency} -->
       <scope>test</scope>
     </dependency>
+    <dependency>
+      <groupId>com.azure</groupId>
+      <artifactId>azure-core-http-netty</artifactId>
+      <version>1.14.2</version> <!-- {x-version-update;com.azure:azure-core-http-netty;dependency} -->
+      <scope>compile</scope>
```
Comment (MichaelZp0) [inline sdk/remoterendering/azure-mixedreality-remoterendering/pom.xml:101]: We probably only need a test scope here.

Comment (MichaelZp0) [reply sdk/remoterendering/azure-mixedreality-remoterendering/pom.xml:101]: There is a discussion in the original fix https://github.com/Azure/azure-sdk-for-java/pull/39587 about exactly this, but I don't think we even use netty outside of our tests, so we should be able to use test scope.

Comment (MichaelZp0) [reply sdk/remoterendering/azure-mixedreality-remoterendering/pom.xml:101]: Discussed offline and just to be safe when also including other projects we'll keep it on compile.

Comment (ChristopherManthei) [reply sdk/remoterendering/azure-mixedreality-remoterendering/pom.xml:101]: Internal communication says `compile` scope and I'm just following other packages by example.

**Pull request merged**